### ISSUE TITLE
[Question] GPG encryption on Telegram

### ISSUE THREAD

**Issue by PandaFoss** [Telegram-FOSS-Team/Telegram-FOSS#393]
Hello! I don't know much about encryption, but Telegram is often questioned as not really private and secure since it doesn't have end-to-end encryption in default chats and groups. My question is, is it possible to add to the client the option to encrypt messages using GPG? I can think of something similar to how GPG is used in email (with Enigmail and Thunderbird for example). That is, you can share the public key and then sign and encrypt messages from a non-secret chat. I hope it's not such a crazy idea! :)

**Comment by radasbona** [comment]
It's a great idea imo.
Like conversations can use openkeychain for pgp encryption. Brings multi device e2ee into telegram for android.

**Comment by PandaFoss** [comment]
I have also seen that applications like Tessercube intend to bring OpenPGP to most mobile applications. They can be used as options for encrypting messages on Telegram using OpenPGP, but it would be much more practical and less invasive for the client to include the encryption option as I suggest in the first message.

**Comment by radasbona** [comment]
Tessercube crashes everytime when i want to encrypted something. Oversec works, but looks very ugly.

Thats why i love your idea of a telegram client with build in pgp feature. No more need to use the very good hidden secret chats.

**Comment by rustforfuture** [comment]
Yes, apps like oversec that are based in accessibility api can't use in telegram because telegram won't honor and show its app content and texts to accessibility apps.

This is great feature if you add it.

Thanks.
